### DISCUSSION TITLE
Hörmann Rollmatic v2 working like a charm!

### DISCUSSION THREAD

**Discussion by phil-jas** [Gifford47/HCPBridgeMqtt#38] [Show and tell]
Recently a [Hörmann Rollmatic v2 garage door](https://www.hoermann.de/mediacenter/download/264282de/4510873_RollMatic_2_DE.pdf?20230806132122) was installed and since I wanted to control it via Home Assistant, I gave this project a try - hardware setup (ESP32 + DC-DC + RS485) and code (from dev branch) worked like a charm!

@Gifford47 while implementing this project, I collected some useful information which might be added to the readme.md, in case someone else with Rollmatic v2 would give it a try.

**Compatibility:**

- add Rollmatic v2 as compatible

**Procedure to perform a bus scan for Rollmatic V2:**

_Hardware part:_
1. Remove power cord
2. Plug in RJ-11 connector
3. Plug-in power cord
4. Drive the garage door completely up or down once

_Programming:_
1. Hold "PGR" button until 10 is shown
2. Press "Down" button until "37" is shown
3. Press "PRG" button once -> Display shows "00" (meaning "no external modules configured")
4. Press "Up" button once until 01 is shown (meaning "configure external module")
5. Hold "PGR" button until "B.S." is shown
6. When "B.S." starts blinking, bus scan is running (RX and TX on RS485 board active)
7. If external module was recognized on the bus, the display shows "1" for a couple of seconds
8. If external module was not recognized on the bus, the display shows "01" or "00" for a couple of seconds
9. Menu goes back to "10"

_Troubleshooting Error "04":_
Hörmann Rollmatic v2 uses HCP2-Bus which requires external modules to be registered and unregistered via bus scan. Error "04" shows up when you want to drive the door, but the registered external module was removed without performing a bus scan afterwards. The door will not drive until the issue is solved.

This can be archieved by Powering off the door, re-attach the external module and power on again.
Or by performing a complete factory reset of the Rollmatic v2.

Unfortunately a simple un-registration of a registered (but non working or non-attached module) seems not possible since in my case, Error "04" prevents access to the programming menu after powering on...

Best regards,
Phil

**Comment by Gifford47** [comment]
Hello @phil-jas ,
thanks for your investigation! I will add it to the next readme release soon!

best regards,
gifford
